### PR TITLE
ci(goreleaser): fix release candidate suffix

### PR DESCRIPTION
**Pull request by rootulp**
Prevents https://github.com/celestiaorg/celestia-app/issues/3819 from happening again. Also adds a document so other maintainers can generate + upload prebuilt binaries if they don't work again in the future.

The current `-rc*` [prerelease_suffix](https://goreleaser.com/customization/git/#git) was not working because goreleaser treated v2.1.2-rc1 as more recent than v2.1.2 in the [failing workflow](https://github.com/celestiaorg/celestia-app/actions/runs/10523365886/job/29157983992#step:6:68). This resolves by removing the `*`.

## Testing

1. Worked on my fork: https://github.com/rootulp/celestia-app/releases/tag/v2.1.2
2. Worked on this repo: https://github.com/celestiaorg/celestia-app/releases/tag/v2.1.2

### Before

```
    • git state                                      commit=48173df3dc78f9348eedb3796f29ef9e9e5dc584 branch=HEAD current_tag=v2.1.2-rc1 previous_tag=v2.1.2-rc0 dirty=true
```
### After

```
    • git state                                      commit=48173df3dc78f9348eedb3796f29ef9e9e5dc584 branch=HEAD current_tag=v2.1.2 previous_tag=v2.1.2-rc0 dirty=true
```

It is curious that it thinks the previous tag is v2.1.2-rc0 and not v2.1.2-rc1 but I couldn't fix with various [tag_sort](https://goreleaser.com/customization/git/?h=tag_sort#git) options. I think this only impacts Goreleaser changelog generation which we don't use b/c we use Github's auto generate release notes feature.

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -118,4 +118,5 @@ changelog:
 release:
   prerelease: auto
 git:
-  prerelease_suffix: "-rc*"
+  prerelease_suffix: "-"
```
Comment (rootulp) [inline .goreleaser.yaml:121]:
```suggestion
  prerelease_suffix: "-rc"
```

File: .goreleaser.yaml [hunk 2]
```diff
@@ -118,4 +118,5 @@ changelog:
 release:
   prerelease: auto
 git:
-  prerelease_suffix: "-rc*"
+  prerelease_suffix: "-rc"
+  tag_sort: semver
```
Comment (rootulp) [inline .goreleaser.yaml:122]:
```suggestion
  tag_sort: -version:refname
```

File: .goreleaser.yaml [hunk 3]
```diff
@@ -118,4 +118,5 @@ changelog:
 release:
   prerelease: auto
 git:
-  prerelease_suffix: "-rc*"
+  prerelease_suffix: "-rc"
+  tag_sort: -version:refname
```
Comment (rootulp) [inline .goreleaser.yaml:122]:
```suggestion
```

**Review by cmwaters — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]
utAck

**Pull request merged**